evap/static/ts/src/csrf-utils.ts

```diff
@@ -28,6 +28,15 @@ $.ajaxSetup({
     },
 });
 
+var inputElement = document.createElement("input");
+inputElement.setAttribute("type", "hidden");
+inputElement.setAttribute("name", "csrfmiddlewaretoken");
+inputElement.setAttribute("value", csrftoken);
+
+for (const key of Array.from(document.getElementsByTagName("form"))) {
+    key.insertAdjacentHTML("afterbegin", inputElement.outerHTML);
+}
```

Comment:
I think the following should be equivalent:

```suggestion
for (const form of document.forms) {
    form.insertAdjacentElement("afterbegin", inputElement);
}
```

I am unsure if it is bad to insert the same element multiple times, I think not, but then there is `inputElement.cloneNode() as Element`. Don't know, maybe serializing and deserializing is best after all :thinking:

Reply:
insert seems to move, not copy, the element. so the clone is actually necessary.